### PR TITLE
New serverless pattern for cloudwatch metric streams to kinesis firehose

### PR DESCRIPTION
**Pull request by kiranram**
*Issue #, if available:*

*Description of changes:*
This new pattern creates CloudWatch Metric Streams, which is sent to Kinesis Firehose with S3 bucket as destination. In this pattern the region is set to eu-west-2 but can be configured. The implementation is in Terraform.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by biswanathmukherjee** [comment]
Please add `example-pattern.json` as provided in the sample here: https://github.com/aws-samples/serverless-patterns/tree/main/_pattern-model

**Comment by kiranram** [comment]
The example-pattern.json file has been added now.

**Comment by kiranram** [comment]
Updated to Amazon Data Firehose

**Review by biswanathmukherjee — CHANGES_REQUESTED** [review]

File: cloudwatch-metric-streams-firehose-terraform/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    ```
+
+## How it works
+When AWS services are provisioned, the listed metrics(in the IaC) will be captured and streamed to Kinesis Firehose. The destination in this case is a S3 bucket, where the metrics are saved. The code is configured to eu-west-2, but can be changed to any desired region.
+
+![pattern](Images/pattern.png)
+
+## Testing
+
+After deployment, launch an EC2 instance in eu-west-2 region, and after a few minutes the metrics data will appear in the S3 bucket.
```
[the remark this answers is not in the document]
Comment (kiranram) [reply cloudwatch-metric-streams-firehose-terraform/README.md]: The README file has been updated on how to use CLI to override the default region. This should allow having streams in any region as per the users choice.

The file is stored in Amazon S3 in GZIP format, so using gunzip should extract the contents. The default format for contents was set to Opentelemetry0.7, which has now been changed to JSON (human readable). The metrics written into these files are as per what has been configured in the IaC (main.tf), which can be altered/modified as per the requirements. In this example, there are 2 metrics (CPU utilization and NetworkOut) from EC2 (or RDS if provisioned) which appear in the files.

File: cloudwatch-metric-streams-firehose-terraform/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    "text": [
+      "terraform destroy"
+    ]
+  },
+  "authors": [
+    {
+      "name": "Kiran Ramamurthy",
+      "image": "n/a",
+      "bio": "I am a Senior Partner Solutions Architect for Enterprise Transformation. I work predominantly with partners and specialize in migrations and modernization.",
+      "linkedin": "https://www.linkedin.com/in/kiran-ramamurthy-a96341b/",
```
[the remark this answers is not in the document]
Comment (kiranram) [reply cloudwatch-metric-streams-firehose-terraform/example-pattern.json]: Updated with ID.

**Pull request merged**